src/components/Layout.js

```diff
@@ -2,10 +2,26 @@ import React from 'react'
 import { Link } from 'gatsby'
 
 import { rhythm, scale } from '../utils/typography'
+import { loadState, saveState } from '../utils/helpers'
 
 class Layout extends React.Component {
+  constructor() {
+    super()
+    this.state = {
+      lightMode: loadState('lightMode') === undefined || loadState('lightMode'),
```

Comment:
It would be cool if dark mode was enabled [if the user prefers it](https://paulmillr.com/posts/using-dark-mode-in-css/). Something like

```suggestion
      lightMode: loadState('lightMode') || (window.matchMedia && window.matchMedia('(prefers-color-scheme: dark)').matches) || true,
```

Maybe worth a utility function? You could also go all out and add a listener to the `matchMedia` query, so it updates live.